### DISCUSSION TITLE
TypeError: fetch failed, IPv6 problem

### DISCUSSION THREAD

**Discussion by ExorTek** [vercel/next.js#54359] [Help] [answered]
### Summary

TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async invokeRequest (C:\Users\user\Desktop\github\game\node_modules\next\dist\server\lib\server-ipc\invoke-request.js:17:12)
    at async invokeRender (C:\Users\user\Desktop\github\game\node_modules\next\dist\server\lib\router-server.js:254:29)
    at async handleRequest (C:\Users\user\Desktop\github\game\node_modules\next\dist\server\lib\router-server.js:447:24)
    at async requestHandler (C:\Users\user\Desktop\github\game\node_modules\next\dist\server\lib\router-server.js:464:13)
    at async Server.<anonymous> (C:\Users\user\Desktop\github\game\node_modules\next\dist\server\lib\start-server.js:117:13) {
  cause: Error: connect ECONNREFUSED ::1:65233
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 65233
  }
}

**Comment by AdmiJW** [comment, marked as answer]
Same issue here with identical error message. This started occurring when I have decided to upgrade my `next` version from `13.4.10` to `13.4.19` today. I tried to rollback the version to `13.4.10` and the issue is gone again.

Since I was having the same issue as this https://github.com/vercel/next.js/discussions/53352 at version `13.4.10`, I then upgraded the `next` version to `13.4.12`. No problem there as well.

**Reply by ExorTek** [reply]
Yes, as you mentioned, I had this problem with version `13.4.19` Before reverting to the old version `"dev": "next dev --hostname 127.0.0.0.1"` but it worked on some devices and not on others.

**Reply by AdmiJW** [reply]
Hi again, updating my situation so far.

In my case, I had come across this [issue](https://github.com/vercel/next.js/issues/46756#issuecomment-1687027844) which kind of explained my situation. My application suffered from frequent `The server is running out of memory, restarting to free up memory` issues and also had slow compilation time, so I tried to optimize the two packages that I am using - Material UI `@mui` and `react-icons` using [this](https://github.com/react-icons/react-icons/issues/786#issuecomment-1674964835) and [this](https://github.com/vercel/next.js/discussions/37614#discussioncomment-3036716). 

Now, not only is the compilation time faster in development, but the `fetch failed` error now occurs much less frequently, albeit it still happens.

**Reply by ExorTek** [reply]
`The server is running out of memory, restarting to free up memory`  I had the same problem. I change next.config.js file:
```
 /** @type {import('next').NextConfig} */
const path = require('path');
const nextConfig = {
 experimental: { appDir: true },
 webpack: config => {
  config.ignoreWarnings = [{ module: /node_modules/ }, { file: /node_modules/ }];
  return config;
 },
 swcMinify: true,
 compress: true,
 poweredByHeader: false,
 sassOptions: {
  includePaths: [path.join(__dirname, 'styles')],
 },
 optimizeFonts: true,
 reactStrictMode: true,
};

module.exports = nextConfig;
```

these settings worked for me.

**Reply by ExorTek** [reply]
I removed what I had done and did it the way you mentioned but it didn't work for me. maybe i can't set next.config.js
``` Import trace for requested module:
./node_modules/pino/lib/tools.js
./node_modules/pino/pino.js
./node_modules/@walletconnect/logger/dist/cjs/index.js
./node_modules/@walletconnect/sign-client/dist/index.cjs.js
./node_modules/@jnwng/walletconnect-solana/lib/esm/index.js
./node_modules/@solana/wallet-adapter-walletconnect/lib/esm/adapter.js
./node_modules/@solana/wallet-adapter-walletconnect/lib/esm/index.js
./node_modules/@solana/wallet-adapter-wallets/lib/esm/index.js
./src/layouts/GlobalLayout.js
- wait compiling...
bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)
- warn ./node_modules/@walletconnect/keyvaluestorage/dist/cjs/node-js/db.js
Module not found: Can't resolve 'lokijs' in '\node_modules\@walletconnect\keyvaluestorage\dist\cjs\node-js'

Import trace for requested module:
./node_modules/@walletconnect/keyvaluestorage/dist/cjs/node-js/db.js
./node_modules/@walletconnect/keyvaluestorage/dist/cjs/node-js/index.js
./node_modules/@walletconnect/keyvaluestorage/dist/cjs/index.js
./node_modules/@walletconnect/core/dist/index.cjs.js
./node_modules/@walletconnect/sign-client/dist/index.cjs.js
./node_modules/@jnwng/walletconnect-solana/lib/esm/index.js
./node_modules/@solana/wallet-adapter-walletconnect/lib/esm/adapter.js
./node_modules/@solana/wallet-adapter-walletconnect/lib/esm/index.js
./node_modules/@solana/wallet-adapter-wallets/lib/esm/index.js
./src/layouts/GlobalLayout.js

./node_modules/pino/lib/tools.js
Module not found: Can't resolve 'pino-pretty' in '\node_modules\pino\lib'

Import trace for requested module:
./node_modules/pino/lib/tools.js
./node_modules/pino/pino.js
./node_modules/@walletconnect/logger/dist/cjs/index.js
./node_modules/@walletconnect/sign-client/dist/index.cjs.js
./node_modules/@jnwng/walletconnect-solana/lib/esm/index.js
./node_modules/@solana/wallet-adapter-walletconnect/lib/esm/adapter.js
./node_modules/@solana/wallet-adapter-walletconnect/lib/esm/index.js
./node_modules/@solana/wallet-adapter-wallets/lib/esm/index.js
./src/layouts/GlobalLayout.js
```

**Reply by ExorTek** [reply]
Exam
```
modularizeImports: {
  '@solana': {
   transform: '@solana/{{member}}',
  },
 },
```

**Reply by ExorTek** [reply]
Worked for me
```
webpack: config => {
  config.externals.push('pino-pretty', 'lokijs', 'encoding');
  return config;
 },
```
